enumeration value 'binary' not handled in switch

Just making sure someone saw this warning

```
/builddir/build/BUILD/jsonnet-0.19.1/core/vm.cpp:1665:16: warning: enumeration value 'binary' not handled in switch [-Wswitch]
 1665 |         switch (v.type()) {
      |                ^
```

--- a/core/vm.cpp
+++ b/core/vm.cpp
@@ -21,6 +21,10 @@ limitations under the License.
 #include <set>
 #include <string>
 
+// charconv is because the RapidYAML 0.5.0 release has a bug in the single-header build.
+// https://github.com/biojppm/rapidyaml/issues/364#issuecomment-1536625415
+#include <charconv>
+
 #include "desugarer.h"
 #include "json.h"
 #include <nlohmann/json.hpp>

--- a/core/vm.cpp
+++ b/core/vm.cpp
@@ -1715,6 +1715,14 @@ class Interpreter {
             case json::value_t::discarded: {
                 abort();
             }
+
+            default: {
+                // Newer mlohmann json.hpp (from v3.8.0 https://github.com/nlohmann/json/pull/1662)
+                // add a `value_t::binary` type, used when dealing with some JSON-adjacent binary
+                // formats (BSON, CBOR, etc). Since it doesn't exist prior to v3.8.0 we can't match
+                // on it explicitly, but we can just treat any unknown type as an error.
+                abort();
+            }
         }
     }
 

